### PR TITLE
DOCS-904: WN edits and nav changes for RN too

### PR DESCRIPTION
**Pull request by oliverhowell**

### CODE REVIEW

File: docs/modules/ROOT/pages/whats-new.adoc [hunk 1]
```diff
@@ -4,67 +4,76 @@
 
 {description}
 
-NOTE: For new and improved features in Hazelcast Platform 5.5, see xref:{page-latest-supported-hazelcast}@hazelcast:ROOT:whats-new.adoc[What's new in Hazelcast Platform 5.5].
+IMPORTANT: Hazelcast Management Center 5.5 requires Java 17 or later.
+
+// Is this recent change? We don't have in 5.4 WN
```
Comment (jgardiner68) [inline docs/modules/ROOT/pages/whats-new.adoc:9]: It was true for 5.4 as well -- mentioned in the whats new for 5.4 but a bit hidden under "Upgrade Requirements"

Comment (oliverhowell) [reply docs/modules/ROOT/pages/whats-new.adoc:9]: Ah, I see it. Probably better as an upfront Important as I've placed it. WDYT?

File: docs/modules/ROOT/pages/whats-new.adoc [hunk 2]
```diff
@@ -4,67 +4,76 @@
...
+Hazelcast supports two kinds of members: 
 
-With Platform 5.5, it's possible to execute Jet jobs only on the Lite members and therefore scale computational power independently of storage. Hazelcast has always supported promoting a Lite member to a Full member and in this release you can now demote a Full member to a Lite member. Management Centre supports this functionality.
+* **Full Members** for when you need to scale both data storage and computational capabilities across the cluster
+* **Lite Members** are useful when you need additional computational power or clients that access data without the overhead of data storage and synchronization
 
-== Multi-version Rolling Upgrades
+With Hazelcast Platform 5.5, it's now possible to execute Jet jobs only on the Lite members and therefore scale computational power independently of storage. Hazelcast has always supported promoting a Lite member to a Full member but in this release you can also now demote a Full member to a Lite member. Management Center supports this feature.
```
Comment (jgardiner68) [inline docs/modules/ROOT/pages/whats-new.adoc:22]:
```suggestion
With Hazelcast Platform 5.5, it's now possible to execute Jet jobs only on the Lite members and therefore scale computational power independently of storage. Hazelcast has always supported promoting a Lite member to a Full member but in this release you can also now demote a Full member to a Lite member. Management Center previously supported upgrading a lite member to full within the Members section of the UI and has been extended to support downgrading a full member to a lite member.
```

Comment (oliverhowell) [reply docs/modules/ROOT/pages/whats-new.adoc:22]:
```suggestion
With Hazelcast Platform 5.5, it's now possible to execute Jet jobs only on the Lite members and therefore scale computational power independently of storage. Management Center has always supported promoting a Lite member to a Full member but in this release you can also now demote a Full member to a Lite member. 
```

**Pull request merged**